src/MongoDBServiceProvider.php

```diff
@@ -40,5 +45,28 @@ public function register()
                 return new MongoConnector($this->app['db']);
             });
         });
+
+        // Add cache store.
+        $this->app->resolving('cache', function (CacheManager $cache) {
+            $cache->extend('mongodb', function (Application $app, array $config): Repository {
+                $connection = $app['db']->connection($config['connection'] ?? null);
+
+                $store = new MongoStore(
+                    $connection,
+                    $config['collection'] ?? $config['table'] ?? throw new InvalidArgumentException('Missing "collection" name for MongoDB cache'),
```

Comment:
Both `collection` and `table` configs are accepted for compatibility with the `database` config. But I may remove the `table` support to be cleaner.

This needs to be consistent with `lock_table` below.

Reply:
Moved to `collection` and `lock_collection` with default values `cache` and `cache_locks`.